src/client/App.jsx

```diff
@@ -1,56 +1,119 @@
+import UserForm from "./components/UserForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
 
 function App() {
-  const [movies, setMovies] = useState([]);
+  const [movies, setMovies] = useState(null);
+  const [updateMovieList, setUpdateMovieList] = useState(0);
+
+  const navigate = useNavigate();
 
   useEffect(() => {
-    fetch(`${apiUrl}/movie`)
-      .then(res => res.json())
-      .then(res => setMovies(res.data));
-  }, []);
+    fetch(`${apiUrl}/movie`, {
+      headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+    })
+      .then((res) => res.json())
+      .then((res) => setMovies(res.data));
+  }, [updateMovieList]);
 
   const handleRegister = async ({ username, password }) => {
-    
+    const url = "http://localhost:4000/user/register";
+    const opts = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ username, password }),
+    };
+
+    const fetchRes = await fetch(url, opts);
+    const data = await fetchRes.json();
+
+    if (data.error) {
+      alert(data.error);
+    }
   };
 
   const handleLogin = async ({ username, password }) => {
-    
+    const url = "http://localhost:4000/user/login";
+    const opts = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ username, password }),
+    };
+
+    const fetchRes = await fetch(url, opts);
+    const data = await fetchRes.json();
+
+    localStorage.setItem("token", data.data);
+
+    if (data.error) {
+      alert(data.error);
+    }
+
+    if (data.data) {
+      setUpdateMovieList(updateMovieList + 1);
+      navigate("/movies");
+    }
   };
-  
+
   const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
-  }
+    const url = "http://localhost:4000/movie";
+    const opts = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${localStorage.getItem("token")}`,
+      },
+      body: JSON.stringify({ title, description, runtimeMins }),
+    };
 
-  return (
-    <div className="App">
-      <h1>Register</h1>
-      <UserForm handleSubmit={handleRegister} />
+    const fetchRes = await fetch(url, opts);
 
-      <h1>Login</h1>
-      <UserForm handleSubmit={handleLogin} />
+    if (data.error) {
+      alert(data.error);
+    }
 
-      <h1>Create a movie</h1>
-      <MovieForm handleSubmit={handleCreateMovie} />
+    if (data.data) {
+      setUpdateMovieList(updateMovieList + 1);
+    }
+  };
 
-      <h1>Movie list</h1>
+  return (
+    <div className="App">
       <ul>
-        {movies.map(movie => {
-          return (
-            <li key={movie.id}>
-              <h3>{movie.title}</h3>
-              <p>Description: {movie.description}</p>
-              <p>Runtime: {movie.runtimeMins}</p>
-            </li>
-          );
-        })}
+        <Link to="/">
```

Comment:
I'd likely expect the set of links here to be a component for the navbar or menu or something similar

Reply:
Links are part of a nav bar ✔️